systemvm/debian/opt/cloud/bin/cs/CsDhcp.py

```diff
@@ -114,8 +114,23 @@ def configure_server(self):
             idx += 1
 
     def delete_leases(self):
+        macs_dhcphosts = []
         try:
-            open(LEASES, 'w').close()
+            logging.info("Attempting to delete entries from dnsmasq.leases file for VMs which are not on dhcphosts file")
+            for host in open(DHCP_HOSTS):
+                macs_dhcphosts.append(host.split(',')[0])
+
+            removed = 0
+            for leaseline in open(LEASES):
+                lease = leaseline.split(' ')
+                mac = lease[1]
+                ip = lease[2]
+                if mac not in macs_dhcphosts:
+                    cmd = "dhcp_release %s %s" % (ip, mac)
```

Comment:
@nvazquez the dhcp_release `<interface>` parameter is missing here

See https://manpages.debian.org/stretch/dnsmasq-utils/dhcp_release.1.en.html

Reply:
Thanks I'll fix this

Reply:
Fixed @gildegoma @nvazquez